modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/model/property/DefaulEncodingModelProperty.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.model.property;
+
+import org.mule.runtime.api.meta.model.ModelProperty;
+import org.mule.runtime.api.meta.model.parameter.ParameterModel;
+
+/**
+ * A {@link ModelProperty} intended to be used on {@link ParameterModel parameters}
+ * to signal that if the parameter is resolved to {@code null}, then the runtime should
```

Comment:
careful here. One thing is `resolves to null` and another thing is `is not provided`. What you did is not about resolving to null , but about not being provided. Two very different things. If you want to make this about not resolving to null, then you need to wrap the resolver instead of using a static one like you did. I think you should allow values resolving to null. Users might want to default to different stuff, like email does.